cli: simplify resetting commands (#8312)
tendermint/tendermint: UPGRADING.md

For more discussion of this, see [ADR 074](https://tinyurl.com/adr074), which
lays out the reasoning for the changes as well as [RFC
009](https://tinyurl.com/rfc009) for a discussion of the complexities of

> upgrading consensus parameters.\

Author: ```suggestion
upgrading consensus parameters.
```

### CLI Changes

> The functionality around resetting a node has been extended to make it safer. The CLI
> now has four `reset` subcommands: `blockchain`, `peers`, `unsafe-signer` and `unsafe-all`
> in replacement of the previous `unsafe-reset-all`.

Reviewer: ```suggestion
The functionality around resetting a node has been extended to make it safer. The
`unsafe-reset-all` command has been replaced by a `reset` command with four
subcommands: `blockchain`, `peers`, `unsafe-signer` and `unsafe-all`.
```

> - `tendermint reset blockchain`: clears a node of all blocks, tendermint state, evidence,
>   and indexed transactions. NOTE: this won't affect application state which will need
>   to be separately reset. Please look into `tendermint rollback` in the event that you
>   need to rollback the last state due to application non-determinisim.

Reviewer: ```suggestion
- `tendermint reset blockchain`: Clears a node of all blocks, consensus state, evidence,
  and indexed transactions. NOTE: This command does not reset application state.
  If you need to rollback the last application state (to recover from application
  nondeterminism), see instead the `tendermint rollback` command.
```

Question: When we talk about clearing "indexed transactions" that suggests we are also throwing away the results of ABCI indexing. Is that true? 

If so: Do we need to call out the fact that this only happens with the KV indexer? (We have no way to reset Postgres, if someone happens to be using that—and more generally the indexing API doesn't have a hook for it)

If not: Maybe "stored transactions" to avoid confusion with ABCI indexing?

Author: > When we talk about clearing "indexed transactions" that suggests we are also throwing away the results of ABCI indexing. Is that true?

Yes we're also removing `tx_index.db` if it exists. 

You're right that if someone is using postgres it won't have an affect on it. I could explicitly specify that the kv indexer will be removed if it exists

> - `tendermint reset peers`: clears the peer store which persist information on peers used
>   by the networking layer. This could be important when clearing stale addresses or switching
>   to using a defined set of peers.

Reviewer: ```suggestion
- `tendermint reset peers`: Clears the peer store, which persists information on peers used
  by the networking layer. This can be used to get rid of stale addresses or to switch
  to a predefined set of static peers.
```
